script/constructron.lua

```diff
@@ -1018,86 +1038,94 @@ me.get_job = function()
                 })
 
                 -- request items
-                me.create_job(global.job_bundle_index, {
-                    action = 'request_items',
-                    action_args = {combined_chunks.requested_items},
-                    leave_condition = 'request_done',
-                    constructron = worker,
-                    unused_stations = me.get_service_stations(worker.surface.index)
-                })
+                non_optimistic_item_requested = false
+                for k,item in pairs(combined_chunks.requested_items) do
+                    if not custom_lib.table_has_value(global.optimistic_logistics_items, item) then
+                        non_optimistic_item_requested = true 
+                    end
+                end
 
-                for i, chunk in ipairs(combined_chunks) do
-                    chunk['positions'] = chunk_util.calculate_construct_positions({chunk.minimum, chunk.maximum}, worker.logistic_cell.construction_radius * 0.95) -- 5% tolerance
-                    chunk['surface'] = surface.index
-                    for p, position in ipairs(chunk.positions) do
+                if non_optimistic_item_requested then

```

Comment:
why is this check needed? why do we need to change the way jobs get created & bundled for this feature?

Reply:
basically it says "dont create this as a job if the only items it wants are optimistic" because that would mean the spidertrons would move from a to b in a loop infinitely.

Reply:
Yup absolutely, obviously this PR was meant to just be a first attempt to see the usecase of it. Ive been using this fork on a server for the last few days and have found it really beneficial! But long term avoiding looping needs to be done more intelligently.